### ISSUE TITLE
Add support to cascade all entities.

### ISSUE THREAD

**Issue by mshima** [jhipster/generator-jhipster#12844]
##### **Overview of the feature request**

Value object are really useful to keep code clean and maintainable.
This feature request proposes the implementation `@valueObject(cascade)`.
- A cascade all entity is not a real value object, because it has identity, but the implementation is related that's why IMO we should implement it as `@valueObject(cascade)`.

Once implemented we could add others value objects implementations:
- `@valueObject(embedded)` using `@Embbeded` annotation. (There is already a `@Embedded` implementation for backend-only and cassandra database).
- `@valueObject(json)` converting the object into a json string when persisting, (use jsonp type in postgres).

Scope of this feature request:
- `@valueObject(cascade)` backend implementation.
- angular frontend implementation.
- cypress for value object forms will be kept out for now.

Cascade implementation:
- Cascade removes backend apis related to list/find while value objects should remove every backend api specific to the object.

##### **Related issues or PR**

-   [ ] Checking this box is mandatory (this is just to show you read everything)

**Comment by DanielFran** [comment]
@mshima This has been reopened, but I am unsure if this implementation is useful?

**Comment by xinlmain** [comment]
Yes please! Value Objects are essential for DDD, without this we have to write a lot of code, to alter what JHipster generated.

See: 
https://github.com/jhipster/jhipster-core/issues/262
https://stackoverflow.com/questions/41082676/generate-value-object-not-entity-with-jdl-studio
https://github.com/jhipster/jhipster-core/issues/128
